contracts/managers/SocialTradingManager.sol

```diff
@@ -0,0 +1,296 @@
+            _newAllocation,
+            SCALE_FACTOR
+        );
+
+        _tradingPool.startRebalance(address(nextSet));
+
+        emit AllocationUpdate(
+            address(_tradingPool),
+            pools[address(_tradingPool)].currentAllocation,
+            _newAllocation
+        );
+
+        pools[address(_tradingPool)].currentAllocation = _newAllocation;
+    }
+
+    /*
+     * Update trader allowed to manage trading pool.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newTrader          Address of new traders
+     */
+    function setTrader(
+        IRebalancingSetTokenV2 _tradingPool,
+        address _newTrader
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        emit NewTrader(
+            address(_tradingPool),
+            pools[address(_tradingPool)].trader,
+            _newTrader
+        );
+
+        pools[address(_tradingPool)].trader = _newTrader;        
+    }
+
+    /*
+     * Update liquidator used by tradingPool.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newLiquidator      Address of new Liquidator
+     */
+    function setLiquidator(
+        IRebalancingSetTokenV2 _tradingPool,
+        ILiquidator _newLiquidator
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        _tradingPool.setLiquidator(_newLiquidator);
+    }
+
+    /*
+     * Update fee recipient of tradingPool.
+     *
+     * @param _tradingPool          The address of the trading pool being updated
+     * @param _newFeeRecipient      Address of new fee recipient
+     */
+    function setFeeRecipient(
+        IRebalancingSetTokenV2 _tradingPool,
+        address _newFeeRecipient
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        _tradingPool.setFeeRecipient(_newFeeRecipient);
+    }
+
+    /* ============ Internal ============ */
+
+    /*
+     * Validate trading pool allocation update. Make sure trader is caller, allocation is valid,
+     * and RebalancingSet is in valid state.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newAllocation      New base asset allocation in a scaled decimal value
+     *                                          (e.g. 100% = 1e18, 1% = 1e16)
+     */
+    function validateAllocationUpdate(
+        IRebalancingSetTokenV2 _tradingPool,
+        uint256 _newAllocation
+    )
+        internal
+        view
+    {
+        require(
+            _newAllocation <= SCALE_FACTOR,
+            "SocialTradingManager.validateAllocationUpdate: New allocation is not valid."
+        );
+
+        // Require that enough time has passed from last rebalance
+        uint256 lastRebalanceTimestamp = _tradingPool.lastRebalanceTimestamp();
+        uint256 rebalanceInterval = _tradingPool.rebalanceInterval();
+        require(
```

Comment:
Not opposed to these checks being here - though am noting they are on the RBSetV2 already

Reply:
Yeah it's something we did on the previous managers to stop excess gas usage. This way the user doesn't accidentally call it and then is responsible for the gas cost of creating a SetToken